### PR TITLE
Update Documentation, Add End-to-End Script and Instructions

### PR DESCRIPTION
**Pull request by mfehr**
* Moved the README to the root folder
 * Extend the end to end script to include the optimization
 * Simplify communication between dual quaternion based calibration and batch estimation (the json file is now directly exported by the first step)
 * Update README

### CODE REVIEW

**Comment by alexmillane** [comment]
@mfehr This was super useful. Thanks for doing this :+1:

File: README.md [hunk 1]
```diff
@@ -89,39 +89,69 @@ In our hand-eye calibration we use the following frames:
...
+    --time_offset_output_csv_file time_offset.csv
   ```
 - Perform the dual-quaternion-based hand-eye calibration:
-  ```
-  rosrun hand_eye_calibration compute_hand_eye_calibration.py --aligned_poses_B_H_csv_file tf_aligned.csv --aligned_poses_W_E_csv_file camera_aligned.csv --visualize True
+  ```bash
+  rosrun hand_eye_calibration compute_hand_eye_calibration.py \
+    --aligned_poses_B_H_csv_file tf_aligned.csv  \
+    --aligned_poses_W_E_csv_file camera_aligned.csv \
+    --time_offset_output_csv_file time_offset.csv \
```
Comment (HannesSommer) [inline README.md]: Isn't this an input parameter here? Maybe call it input or just `time_offset_csv_file`

Comment (mfehr) [reply README.md]: agreed, this is wrong, will fix it

File: hand_eye_calibration/bin/compute_hand_eye_calibration.py [hunk 2]
```diff
@@ -38,7 +41,18 @@
...
-  parser.add_argument('--visualize', type=bool, default=False, help='Visualize the poses.')
+  parser.add_argument(
+      '--time_offset_output_csv_file', type=str,
+      help='Time offset input file. Is used to construct the calibration json '
+           'file that is needed for the optimization step.')
+  parser.add_argument(
+      '--calibration_output_json_file', type=str,
+      help='Calibration output file. Contains the result of the '
+           'dual-quaternion-based hand eye calibration and time alignment. '
+           'Is ued as a initial guess for the batch estimation.')
```
Comment (mfehr) [inline hand_eye_calibration/bin/compute_hand_eye_calibration.py]: typo

**Comment by mfehr** [comment]
@HannesSommer ptal

**Comment by mfehr** [comment]
test this please (gcc failed for some weird reason, clang might have failed due to recent changes to opencv3_catkin)

**Comment by mfehr** [comment]
test this please (wiped the ws)

**Comment by HannesSommer** [comment]
Ah, it started two builds. Sometimes this happens. One failed (open cv 3 issues) the other succeeded. I've started again too gain more insight.

**Comment by mfehr** [comment]
Build succeeded [here](https://jenkins.asl.ethz.ch/job/hand_eye_calibration/96/), will merge this as the build failures are unrelated to the PR, there seems to be some issue in lynx-xenial. Thanks @HannesSommer for your help!

**Pull request merged**